server/hydrogen-render/1-render-hydrogen-to-string.js

```diff
@@ -23,10 +23,41 @@ if (!logOutputFromChildProcesses) {
   );
 }
 
+function assembleErrorAfterChildExitsWithErrors(exitCode, childErrors) {
+  assert(childErrors);
+
+  let extraErrorsMessage = '';
+  if (childErrors.length > 1) {
+    extraErrorsMessage = ` (somehow we saw ${
+      childErrors.length
+    } errors but we really always expect 1 error)\n${childErrors
+      .map((childError, index) => `${index}. ${childError.stack}`)
+      .join('\n')}`;
+  }
+
+  let childErrorToDisplay;
+  if (childErrors.length === 0) {
+    childErrorToDisplay = new Error('No child errors');
+    childErrorToDisplay.stack = 'No child errors';
+  } else if (childErrors.length === 1) {
+    childErrorToDisplay = childErrors[0];
+  } else {
+    childErrorToDisplay = new Error('Multiple child errors listed above ^');
+  }
+
+  const childErrorSummary = new RethrownError(
+    `Child process exited with code ${exitCode}${extraErrorsMessage}`,
+    childErrorToDisplay
+  );
+
+  return childErrorSummary;
+}
+
 async function renderHydrogenToString(renderOptions) {
+  let abortTimeoutId;
   try {
-    let data = '';
     let childErrors = [];
+    let childExitCode = '(not set yet)';
```

Comment:
Whenever we log `childErrors`, we now also show the `childExitCode` and this is a useful default over `null` or `undefined` which can be a head scratcher whether the code is even doing the right thing.